order.md

```diff
@@ -0,0 +1,41 @@
+# Order
+Ordered source code makes finding stuff faster
+
+## File
+1. Imports
+2. Combined Types (`type Handler<T> = `...)
+3. Supporting small classes (`class PreviewAngle`, bigger classes into own file)
+4. Main exported Class (`export class AngleComponent`)
+5. Default subclasses/implementations (`export class HashRouter`)
+
+## General Classes
+1. Constants (`readonly animationDuration = 2000;`)
+2. Events & Observables (`onPriceUpdate: (price: number) => void;`)
+3. Variables (`animationState: nuumber;`)
+4. Constructor
+5. Public methods
+6. Private methods
+
+## Components
+1. Parent / Parameters (`declare parent: RingEditComponent`)
+2. Constants
+3. Events & Observables (`onPriceUpdate: (price: number) => void;`)
+4. Referenced elements (`canvas: HTMLCanvasElement`)
+5. Variables
+6. Constructor
+7. `onload`
+8. `onunload`
+9. `render`
+10. Sub-renders (`renderGalleryItem(item: RingViewModel) ...`)
+11. Public methods
```

Comment:
Would place public and private methods before render and sub-renders to clearly separate html and typescript.

Currently a component would have ts, html and then again ts...

Reply:
fair point

my goal was to group the component stuff (load & render), as those will be the most important areas to edit